index.js

```diff
@@ -164,124 +157,136 @@ function onRequestResponse (options, error, response, body) {
     return callback(null, response, body);
   }
 
+  // Decompress brotli compressed responses
+  if (/\bbr\b/i.test('' + response.caseless.get('content-encoding'))) {
+    if (!brotli.isAvailable) {
+      const cause = 'Received a Brotli compressed response. Please install brotli';
+      return callback(new RequestError(cause, options, response));
+    }
+
+    response.body = body = brotli.decompress(body);
+  }
+
   if (response.isCloudflare && response.isHTML) {
     onCloudflareResponse(options, response, body);
   } else {
-    processResponseBody(options, response, body);
+    onRequestComplete(options, response, body);
   }
 }
 
 function onCloudflareResponse (options, response, body) {
-  var callback = options.callback;
+  const callback = options.callback;
 
-  var stringBody;
-  var isChallenge;
-  var isRedirectChallenge;
+  let stringBody;
+  let isChallenge;
+  let isRedirectChallenge;
 
   if (body.length < 1) {
     // This is a 4xx-5xx Cloudflare response with an empty body.
-    return callback(new errors.CloudflareError(response.statusCode, options, response));
+    return callback(new CloudflareError(response.statusCode, options, response));
   }
 
   stringBody = body.toString('utf8');
 
   try {
     validate(options, response, stringBody);
   } catch (error) {
+    if (error instanceof CaptchaError && typeof options.onCaptcha === 'function') {
```

Comment:
We need to document this

Reply:
I would add a few examples as a part of that documentation. See the relevant issue for examples.